dotnet/src/Connectors/Connectors.Memory.Qdrant/QdrantVectorStoreRecordCollection.cs

```diff
@@ -499,6 +499,67 @@ public virtual async Task<VectorSearchResults<TRecord>> VectorizedSearchAsync<TV
         return new VectorSearchResults<TRecord>(mappedResults.ToAsyncEnumerable());
     }
 
+    /// <inheritdoc />
+    public async IAsyncEnumerable<TRecord> GetAsync(Expression<Func<TRecord, bool>> filter, int top,
+        FilterOptions<TRecord>? options = null, [EnumeratorCancellation] CancellationToken cancellationToken = default)
+    {
+        Verify.NotNull(filter);
+        Verify.NotLessThan(top, 1);
+
+        options ??= new();
+
+        var translatedFilter = new QdrantFilterTranslator().Translate(filter, this._model);
+
+        // Specify whether to include vectors in the search results.
+        WithVectorsSelector vectorsSelector = new() { Enable = options.IncludeVectors };
+
+        KeyValuePair<Expression<Func<TRecord, object?>>, bool>? sortExpression = options.Sort.Values.Count switch
+        {
+            0 => null,
+            1 => options.Sort.Values[0],
+            _ => throw new NotSupportedException("Qdrant does not support ordering by more than one property.")
+        };
+
+        OrderBy? orderBy = null;
+        if (sortExpression.HasValue)
+        {
+            var orderByName = this._model.GetDataOrKeyProperty(sortExpression.Value.Key) switch
+            {
+                VectorStoreRecordKeyPropertyModel => "id",
```

Comment:
Unless I'm mistaken, the StorageName on the property should already be `id` - the intention here is for StorageName to always contain what the property is actually called in the database. So unless there's some special Qdrant consideration, if you're seeing something else in StorageName, that's probably a bug in model-building...

Reply:
Hmm I wrote it before your refactor, I am going to change it and take advantage of it now.